web/src/main/java/ua/com/vertex/controllers/CourseDetailsController.java

```diff
@@ -0,0 +1,124 @@
+import ua.com.vertex.logic.interfaces.UserLogic;
+
+import javax.validation.Valid;
+import java.util.List;
+
+import static ua.com.vertex.controllers.AdminController.ADMIN_JSP;
+import static ua.com.vertex.controllers.CertificateDetailsPageController.ERROR;
+import static ua.com.vertex.controllers.CreateCertificateAndUserController.MSG;
+
+@Controller
+public class CourseDetailsController {
+
+    static final String COURSE_DETAILS_JSP = "courseDetails";
+    static final String SEARCH_COURSE_JSP = "searchCourse";
+    static final String COURSE_DATA = "courseForInfo";
+    static final String COURSE = "course";
+    static final String COURSES = "courses";
+    private static final String COURSE_ID = "courseId";
+    static final String TEACHERS = "teachers";
+    static final String LOGGER_SERVER_EXCEPTION = "Problems with the server, try again later.";
+
+    private static final Logger LOGGER = LogManager.getLogger(CourseDetailsController.class);
+
+    private final CourseLogic courseLogic;
+    private final UserLogic userLogic;
+
+    @PostMapping(value = "/searchCourseJsp")
+    public ModelAndView searchCourseJsp() {
+        LOGGER.debug("Show search page for courses");
+        return new ModelAndView(SEARCH_COURSE_JSP, COURSE_DATA, new Course());
+    }
+
+    @PostMapping(value = "/searchCourse")
+    public String searchCourse(@Validated @ModelAttribute(COURSE_DATA) Course course,
+                               BindingResult bindingResult, Model model) {
+        LOGGER.debug(String.format("Search user by name - (%s) and finished - (%s).",
+                course.getName(), course.isFinished()));
+        String result = SEARCH_COURSE_JSP;
+
+        if (!bindingResult.hasErrors()) {
+            try {
+                List<Course> courses = courseLogic.searchCourseByNameAndStatus(course);
+                if (courses.isEmpty()) {
+                    model.addAttribute(MSG, String.format("Course with name - '(%s)' not found. " +
+                            "Please check the data and try it again.", course.getName()));
+                } else {
+                    model.addAttribute("courses", courses);
+                }
+            } catch (DataAccessException e) {
+                LOGGER.warn(e);
+                model.addAttribute(MSG, LOGGER_SERVER_EXCEPTION);
+            } catch (Exception e) {
+                LOGGER.warn(e);
+                result = ERROR;
+            }
+        }else {
+            model.addAttribute(MSG, "Enter the correct data !!! ");
+        }
+
+        return result;
+    }
+
+    @PostMapping(value = "/courseDetails")
+    public ModelAndView courseDetails(@RequestParam(COURSE_ID) int courseId) {
+        LOGGER.debug(String.format("Go to the course information page. Course ID -: - (%s)", courseId));
+
+        ModelAndView result = new ModelAndView(COURSE_DETAILS_JSP);
+
+        try {
+            result.addObject(COURSE, courseLogic.getCourseById(courseId).orElseThrow((Exception::new)));
```

Comment:
Not my code. I remember I talked to the guys about this particular situation (I'm not sure, I think it was Slava Klimenko) and pointed out at the same problem — no description in the exception body. Maybe, the problem was rectified in the code that Slava hasn't pushed yet, I don't know.